src/lib.rs

```diff
@@ -135,6 +135,38 @@
 //! # }
 //! ```
 //!
+//! ## Specifying argument types
+//!
+//! There are three types of arguments that can be supplied: short,
+//! long and positional.  Like clap, structopt defaults to creating
+//! positional arguments.  If you want a long argument, you can either
+//! specify "long" to get a long flag generated by converting
+//! `snake_case` field names to `kebab-case`.  You want also specify a
+//! `short` one-character flag, or may manually specify the value of
+//! the `long` flag.  An example may make things clearer:
+//!
+//! ```
+//! # #[macro_use] extern crate structopt;
+//! #[derive(StructOpt)]
+//! struct Opt {
+//!   /// This option can be specified with something like "--foo-option value" or "--foo-option=value"
+//!   #[structopt(long)]
+//!   foo_option: String,
+//!   #[structopt(short = "b")]
+//!   /// This option can be specified with something like "-b value" (but not "--bar-option value")
+//!   #[structopt(long)]
+//!   bar_option: String,
```

Comment:
Really? That's not what I understand reading the code, and that's quite surprising.